lec14-git.Rmd

```diff
@@ -1,4 +1,146 @@
 ---
 title: "Scientific collaboration, Git, and GitHub"
+author: "Elliott Sales de Andrade"
 ---
 
+## Lesson preamble
+
+> ### Lesson objectives:
+>
+> - Practice using the 4 main Git commands and the Git-RStudio interface
+> - Learn how and when to use branches
+> - Learn how to effectively use Pull Requests
+> - Project work
+>
+> ### Lesson outline:
+>
+> - Review basic Git usage (15 min)
+> - Branching in Git (30 min)
+> - Pull requests on GitHub (30 min)
+> - Project work
+>     - Continue working on assignment 7 (20 min)
+
+-----
+
+## Git
+
+### Recap
+
+#### Shell commands
+
+What do these do?
+
+- `ls`
+- `cd`
+- `touch`
+- `echo`
+- `>`
+- `cat`
+
+#### Setup
+
+Follow the getting setup steps from command line:
+
+```bash
+git config --global user.name "Your Name"
+git config --global user.email yourname@example.com
+```
+
+Replace "Your Name", with, well, your first and last name. Replace the fake
+email address with the email address you normally use.
+
+Confirm it works by running:
+
+```bash
+git config --list
+```
+
+#### Basic Git commands
+
+What do these do?
+
+- `git init`
+- `git clone`
+- `git diff`
+- `git add`
+- `git commit`
+- `git log`
+- `git status`
+- `git remote`
+
+## Branching
+
+Git commit history is a [directed acyclic
+graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph) (or DAG), which
+means that every single commit always has at least one 'parent' commit (the
+previous commit(s) in the history), and any individual commit can have multiple
+'children'. This history can be traced back through the 'lineage' or
+'ancestry'.
+
+### What are branches?
+
+Branches are simply a *named pointer* to a commit in the DAG, which
+automatically *move forward* as commits are made. Divergent commits (two
+commits with the same parent) could be considered "virtual" branches. Since
+they are simply pointers, branches in Git are very lightweight.
+
+### Why use them?
+
+- To keep experimental work apart
```

Comment:
"apart"? Like separate?

Reply:
? "apart" is a word?

Reply:
Yes, it just usually has an addition to it. Like "apart from everything else" or "apart from the documents"

Reply:
I'm not sure it's really needed, but we can go with "separate" instead.